Client/qtTeamTalk/preferencesdlg.cpp

```diff
@@ -1005,35 +952,31 @@ void PreferencesDlg::slotSaveChanges()
     if(m_modtab.find(SHORTCUTS_TAB) != m_modtab.end())
     {
 #ifdef Q_OS_WIN32
-        TT_HotKey_Unregister(ttInst, HOTKEY_VOICEACTIVATION);
-        TT_HotKey_Unregister(ttInst, HOTKEY_INCVOLUME);
-        TT_HotKey_Unregister(ttInst, HOTKEY_DECVOLUME);
-        TT_HotKey_Unregister(ttInst, HOTKEY_MUTEALL);
-        TT_HotKey_Unregister(ttInst, HOTKEY_MICROPHONEGAIN_INC);
-        TT_HotKey_Unregister(ttInst, HOTKEY_MICROPHONEGAIN_DEC);
-        TT_HotKey_Unregister(ttInst, HOTKEY_VIDEOTX);
-        TT_HotKey_Unregister(ttInst, HOTKEY_REINITSOUNDDEVS);
-        TT_HotKey_Unregister(ttInst, HOTKEY_SHOWHIDE_WINDOW);
+        for (Hotkeys hk = HOTKEY_PUSHTOTALK; hk < HOTKEY_NEXT_UNUSED; hk <<= 1)
+        {
+            TT_HotKey_Unregister(ttInst, static_cast<HotKeyID>(hk));
+        }
 #endif
-        deleteHotKeySettings(HOTKEY_VOICEACTIVATION);
-        deleteHotKeySettings(HOTKEY_INCVOLUME);
-        deleteHotKeySettings(HOTKEY_DECVOLUME);
-        deleteHotKeySettings(HOTKEY_MUTEALL);
-        deleteHotKeySettings(HOTKEY_MICROPHONEGAIN_INC);
-        deleteHotKeySettings(HOTKEY_MICROPHONEGAIN_DEC);
-        deleteHotKeySettings(HOTKEY_VIDEOTX);
-        deleteHotKeySettings(HOTKEY_REINITSOUNDDEVS);
-        deleteHotKeySettings(HOTKEY_SHOWHIDE_WINDOW);
-
+        for (Hotkeys hk = HOTKEY_PUSHTOTALK; hk < HOTKEY_NEXT_UNUSED; hk <<= 1)
+        {
+            deleteHotKeySettings(static_cast<HotKeyID>(hk));
+        }
         hotkeys_t::iterator ite = m_hotkeys.begin();
-        while(ite != m_hotkeys.end())
+        while (ite != m_hotkeys.end())
         {
             saveHotKeySettings(ite.key(), *ite);
 #ifdef Q_OS_WIN32
             TT_HotKey_Register(ttInst, ite.key(), &(*ite)[0], INT32(ite->size()));
 #endif
             ite++;
         }
+        Hotkeys activeHotkeys = 0;
+        for (auto it = m_shortcutsmodel->getShortcuts().begin(); it != m_shortcutsmodel->getShortcuts().end(); ++it)

```

Comment:
This code is crashing. You get a copy of shortcuts in `.begin()` and another copy of shortcuts in` .end()`. This means that iterators will never match.` getShortcuts(`) has to return a const reference:

`const hotkeys_t& ShortcutsModel::getShortcuts()`